### PR TITLE
fix: check if requested material recieved to create work order in pro…

### PR DESCRIPTION
**Pull request by hitenshKharva**
Ref: [https://app.asana.com/0/1199082161983365/1199559073342222](url)

https://user-images.githubusercontent.com/56474017/102860156-45442400-4453-11eb-98d7-d1d9b908c54a.mp4

### CODE REVIEW

File: erpnext/manufacturing/doctype/production_plan/production_plan.js [hunk 1]
```diff
@@ -121,14 +121,20 @@ frappe.ui.form.on('Production Plan', {
...
+				doc: frm.doc,
+				callback: function() {
+					frm.reload_doc();
+				}
+			});
+		}
+		else {
+            frappe.msgprint("Requested Material not yet received")
+        }
+
```
Comment (vishdha) [inline erpnext/manufacturing/doctype/production_plan/production_plan.js]: Remove this change
```suggestion

```
and 
Add at line number 63
```suggestion
		if (frm.doc.status === 'Material Received') {
			frm.trigger("make_work_order");
		} else {
			frappe.msgprint("Requested Material not yet received")
		}

```

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.js [hunk 2]
```diff
@@ -60,7 +60,12 @@ frappe.ui.form.on('Production Plan', {
 		if (frm.doc.docstatus === 1 && frm.doc.po_items
 			&& frm.doc.status != 'Completed') {
 			frm.add_custom_button(__("Work Order"), ()=> {
-				frm.trigger("make_work_order");
+				if (frm.doc.status==="Material Received") {
+					frm.trigger("make_work_order");
+				}
+				else {
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.js]:
```suggestion
				} else {
```

File: erpnext/manufacturing/doctype/production_plan/production_plan.js [hunk 3]
```diff
@@ -60,7 +60,12 @@ frappe.ui.form.on('Production Plan', {
...
+					frappe.msgprint("Requested Material not yet received")
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.js]:
```suggestion
					frappe.msgprint(__("Requested Material is not yet received."));
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.js [hunk 4]
```diff
@@ -60,7 +60,12 @@ frappe.ui.form.on('Production Plan', {
 		if (frm.doc.docstatus === 1 && frm.doc.po_items
 			&& frm.doc.status != 'Completed') {
 			frm.add_custom_button(__("Work Order"), ()=> {
-				frm.trigger("make_work_order");
+				if (frm.doc.status==="Material Received") {
```
Comment (vishdha) [inline erpnext/manufacturing/doctype/production_plan/production_plan.js]: @hitenshKharva if the Material Request table is empty then it should be allowed to create a work order.  following change irrespective of the material requested or not it still gives validation error for work order

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.js (hunk 4 above)
Comment (hitenshKharva) [reply erpnext/manufacturing/doctype/production_plan/production_plan.js]: @vishdha Yes. Will fix it.

**Review by vishdha — APPROVED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**